net_util/src/queue_pair.rs

```diff
@@ -93,12 +93,12 @@ impl TxVirtio {
                 };
                 if result < 0 {
                     let e = std::io::Error::last_os_error();
-                    queue.go_to_previous_position();
 
                     /* EAGAIN */
                     if e.kind() == std::io::ErrorKind::WouldBlock {
                         warn!("net: tx: (recoverable) failed writing to tap: {}", e);
-                        break;
+                        queue.go_to_previous_position();
+                        return Ok(true);
```

Comment:
This looks good but you could have reused the same format that is used for RxVirtio, by using `break` and `exhausted_descs` mutable variable.